skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,342 @@
+        from skpro.distributions.normal import Normal
+        from skpro.distributions.poisson import Poisson
+        from skpro.distributions.t import TDistribution
+
+        ngboost_dists = {
+            "Normal": Normal,
+            "Laplace": Laplace,
+            "TDistribution": TDistribution,
+            "Poisson": Poisson,
+            "LogNormal": LogNormal,
+        }
+
+        kwargs["index"] = kwargs["mu"].index
+        kwargs["columns"] = kwargs["mu"].columns
+
+        skpro_dist = None
+
+        if self.dist in ngboost_dists:
+            skpro_dist = ngboost_dists[self.dist](**kwargs)
+
+        return skpro_dist
+
+    def _predict_proba(self, X):
+        """Predict distribution over labels for data from features.
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+
+        Parameters
+        ----------
+        X : pandas DataFrame, must have same columns as X in `fit`
+            data to predict labels for
+
+        Returns
+        -------
+        y : skpro BaseDistribution, same length as `X`
+            labels predicted for `X`
+        """
+        X = self._check_X(X)
+
+        # The returned values of the Distributions from NGBoost
+        # are different. So based on that they are split into these
+        # categories of loc,scale,mu and s.
+        # Distribution type | Parameters
+        # ------------------|-----------
+        # Normal            | loc = mean, scale = standard deviation
+        # TDistribution     | loc = mean, scale = standard deviation
+        # Poisson           | mu = mean
+        # LogNormal         | s = standard deviation, scale = exp(mean)
+        #                   |     (see scipy.stats.lognorm)
+        # Laplace           | loc = mean, scale = scale parameter
+
+        dist_params = {
+            "Normal": ["loc", "scale"],
+            "Laplace": ["loc", "scale"],
+            "TDistribution": ["loc", "scale"],
+            "Poisson": ["mu"],
+            "LogNormal": ["scale", "s"],
+        }
+
+        skpro_params = {
+            "Normal": ["mu", "sigma"],
+            "Laplace": ["mu", "scale"],
+            "TDistribution": ["mu", "sigma"],
+            "Poisson": ["mu"],
+            "LogNormal": ["mu", "sigma"],
+        }
+
+        kwargs = {}
+
+        if self.dist in dist_params and self.dist in skpro_params:
+            ngboost_params = dist_params[self.dist]
+            skp_params = skpro_params[self.dist]
+            for ngboost_param, skp_param in zip(ngboost_params, skp_params):
+                kwargs[skp_param] = self._pred_dist(X).params[ngboost_param]
+                if self.dist == "LogNormal" and ngboost_param == "scale":
+                    kwargs[skp_param] = np.log(self._pred_dist(X).params[ngboost_param])
+
+                kwargs[skp_param] = self._check_y(y=kwargs[skp_param])
```

Comment:
can you explain why we need this line? I would not have put it here, but you probably had a good reason to add it?

Reply:
The output parameters returned from the ngboost distribution are `numpy` arrays so if I try to access its index and columns in `kwargs["mu"].index` and `kwargs["mu"].columns` it will give an
 ```
AttributeError: 'numpy.ndarray' object has no attribute 'index'
```
 So I am converting it into a pandas `DataFrame` which is also my return type as mentioned in the docstring for the function using `_check_y()`.

Reply:
but do you still need the index, given that you removed it now?
The index (for `y_pred`) should be taken from `X` and `y`, instead.